### PR TITLE
2.3.1

### PR DESCRIPTION
**Pull request by ocefpaf**
See https://trac.osgeo.org/gdal/wiki/Release/2.3.1-News

### CODE REVIEW

**Comment by ocefpaf** [comment]
@conda-forge-admin, please rerender

**Comment by gillins** [comment]
@ocefpaf Any idea how to restart CircleCI? The `kealib` problem should be fixed by: https://github.com/conda-forge/kealib-feedstock/pull/20. Not sure about the `libkml` errors though - seems to be pulling it from defaults.

**Comment by ocefpaf** [comment]
I just restarted it manually. I'm still puzzled by the Travis-CI failure :-/

**Comment by ocefpaf** [comment]
It really sucks that we get all these packages from defaults even with the new format, `conda-build 3` and channel preference. Very disappointing.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,95 +1,85 @@
...
-    - vc 14  # [win and (py35 or py36)]
+    - libtiff
+    - openjpeg
+    - poppler  # [not win]
+    - proj4
+    - sqlite
+    - xerces-c
+    - zlib  # [not win]
+    - xz
     - libstdcxx-ng  # [linux]
```
Comment (isuruf) [inline recipe/meta.yaml]: You can remove this

Comment (ocefpaf) [reply recipe/meta.yaml]: Yep. Hopefully it will pull less packages from `defaults`.

**Comment by ocefpaf** [comment]
Removing `libstdcxx-ng` did not help much. It keeps pulling `libkml` from `defaults`, with `libcurl` and `libboost`, both names we don't use in `conda-forge` BTW, evem with me forcing our `boost` (that is more up-to-date than that `libboost`). This is a horror show now. These are the packages that `conda` insist on picking from `defaults`:

```
    libboost:        1.65.1-habcd387_4             defaults   
    libcurl:         7.60.0-h1ad7b7a_0             defaults   
    libgcc-ng:       7.2.0-hdf63c60_3              defaults   
    libgfortran:     3.0.0-1                       defaults   
    libgfortran-ng:  7.2.0-hdf63c60_3              defaults   
    libkml:          1.3.0-h9d32c78_3              defaults   
    libnetcdf:       4.6.1-h13459d8_0              defaults   
    libopenblas:     0.2.20-h9ac9557_7             defaults   
    libstdcxx-ng:    7.2.0-hdf63c60_3              defaults   
    numpy-base:      1.14.3-py36h0ea5e3f_1         defaults   
    util-linux:      2.21-0                        defaults
```

Sometime I wish that channel preference would be exactly that, a channel preference, falling back to another channel only if the package was not found on the primary one.

**Comment by isuruf** [comment]
@ocefpaf, even though conda solver has been a pain in the past, this time it's not conda solver's fault. You need to build `libkml` with `boost-cpp==1.67.0`, otherwise `boost-cpp==1.67.0` and conda-forge's `libkml` cannot be installed into the same env.

**Comment by ocefpaf** [comment]
> @ocefpaf, even though conda solver has been a pain in the past, this time it's not conda solver's fault. You need to build `libkml` with `boost-cpp==1.67.0`, otherwise `boost-cpp==1.67.0` and conda-forge's `libkml` cannot be installed into the same env.

I know that and I'm working my way up the stack to do this properly.
However, "not conda solver" fault is relative, if the channel preference would work as a real channel preference we would not need to keep bumping our stack just to stay one step ahead of whatever `defaults` does, including the package renames and splits.

**Comment by isuruf** [comment]
> However, "not conda solver" fault is relative, if the channel preference would work as a real channel preference we would not need to keep bumping our stack just to stay one step ahead of whatever defaults does, including the package renames and splits.

I disagree. libkml from conda-forge and boost-cpp=1.67 is incompatible and therefore the only way to install libkml and boost-cpp=1.67.0 was to install libkml from defaults. Only other possibility is to fail installing libkml and boost-cpp=1.67

**Comment by ocefpaf** [comment]
> I disagree. libkml from conda-forge and boost-cpp=1.67 is incompatible and therefore the only way to install libkml and boost-cpp=1.67.0 was to install libkml from defaults. Only other possibility is to fail installing libkml and boost-cpp=1.67

But that is exactly what I would expect to happen :wink:

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,96 +1,85 @@
...
+    - proj4
+    - sqlite
+    - xerces-c
+    - zlib  # [not win]
+    - xz
   run:
-    - boost 1.66.0  # only need to get conda-forge's libkml instead of defaults
-    - curl >=7.54.1,<8
-    - expat 2.2.*
+    - boost  # only need to get conda-forge's libkml instead of defaults
```
Comment (isuruf) [inline recipe/meta.yaml]: Btw, change this to `boost-cpp` so that python is not required for this C/C++ library

**Comment by ocefpaf** [comment]
Looks like we have the right packages here but OS X is still a no go :unamused:

**Comment by gillins** [comment]
> OS X is still a no go

Can you see what the actual error is? I can't work it out....

**Comment by ocefpaf** [comment]
> Can you see what the actual error is? I can't work it out....

Nope. Let's try to re-start it again...

**Comment by gillins** [comment]
Works fine for me on my Mac. Anyone have other ideas? Should we try CircleCI for OSX?

**Comment by ocefpaf** [comment]
> Works fine for me on my Mac. Anyone have other ideas?

No idea.

> Should we try CircleCI for OSX?

It is worth a shot. I'll try that in my next commit.

**Comment by ocefpaf** [comment]
@gillins CircleCI seems fine, let's try this one!

**Pull request merged**